### ISSUE TITLE
Fix request for pmd7 AvoidUserDataInSharedObjects false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#396]
```
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Getter
@EqualsAndHashCode
@Component
@ConfigurationProperties(prefix = "hazelcast")
public class HazelcastProperties implements BaseHazelcastProperties {
    private static final String HAZELCAST = "hazelcast.";
    @Value("${"+ HAZELCAST + "mapNameAccount}")
    private String mapNameAccount; // <--- false positive
}
```

**Comment by jborgers** [comment]
@ConfigurationProperties and no @Setter should be ok
